StreetHawk/Classes/Core/Publish/SHUtils.m

```diff
@@ -750,6 +750,41 @@ BOOL shIsSDKViewController(UIViewController * vc)
             || [vc isKindOfClass:[SHBaseCollectionViewController class]]);
 }
 
+NSString* findNativeID(UIView *view)
+{
+    __block NSString *nativeid = nil;
+    [view.subviews enumerateObjectsWithOptions:NSEnumerationReverse usingBlock:^(__kindof UIView * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+        id nid = [obj valueForKey:@"nativeID"];
+        if (nid && [nid isKindOfClass:NSString.class]) {
+            nativeid = (NSString*)nid;
+        }
+        if (nativeid.length <= 0) {
+            nativeid = findNativeID(obj);
+        }
+        if (nativeid.length > 0) {
+            *stop = YES;
+        }
+    }];
+    return nativeid;
+}
+
+NSString *shAppendUniqueSuffix(UIViewController *vc)
+{
```

Comment:
how is this guranteed to be unique ?

Reply:
If it's react-native, the reactive-native-id is unique. By appending the id as suffix, it's unique.

Reply:
For non-react-native, the vc's class is unique for each page.